### PR TITLE
LwM2M Wi-Fi based location assistance

### PR DESCRIPTION
**Pull request by jarlamsa**
Add support for using Wi-Fi Access points for location.
After scanning, the AP information is added to Visible Wi-Fi Access Point lwm2m objects (ID 33627).
The object data is sent to lwm2m-server with the ground fix assistance request to request ground fix from nRFCloud in a similar way than is done with the cell based request.

### CODE REVIEW

File: samples/nrf9160/lwm2m_client/overlay-assist-wifi.conf [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+CONFIG_MBEDTLS=n
+CONFIG_NORDIC_SECURITY_BACKEND=n
+
+# Networking
+CONFIG_NET_L2_ETHERNET=y
+#CONFIG_NET_SHELL=y
+CONFIG_NET_NATIVE=y
+CONFIG_NET_IPV4=y
+CONFIG_NET_DHCPV4=y
+CONFIG_NET_STATISTICS=y
```
Comment (SeppoTakalo) [inline samples/nrf9160/lwm2m_client/overlay-assist-wifi.conf:63]: Do we really need full IP stack for just scanning?

**Review by SeppoTakalo — APPROVED** [review]

File: samples/nrf9160/lwm2m_client/overlay-assist-wifi.conf (hunk 1 above)
Comment (jarlamsa) [reply samples/nrf9160/lwm2m_client/overlay-assist-wifi.conf:63]: Not actually sure, I'll try to clean some of the configs from the overlay.

File: subsys/net/lib/lwm2m_client_utils/location/location_wifi_ap_scanner.c [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	case NET_EVENT_WIFI_SCAN_RESULT:
+		lwm2m_wifi_scan_result_handle(cb);
+		break;
+	case NET_EVENT_WIFI_SCAN_DONE:
+		lwm2m_wifi_scan_done_handle(cb);
+		break;
+	default:
+		break;
+	}
+
```
Comment (rlubos) [inline subsys/net/lib/lwm2m_client_utils/location/location_wifi_ap_scanner.c:92]: Unneded empty line.

**Review by rlubos — APPROVED** [review]

File: samples/nrf9160/lwm2m_client/overlay-assist-wifi.conf (hunk 1 above)
Comment (jarlamsa) [reply samples/nrf9160/lwm2m_client/overlay-assist-wifi.conf]: Cleaned the overlay file. There are currently some issues with the external flash when using the EK-board over nrf91, both are using the same spi and causes some issues.

**Review by VeijoPesonen — APPROVED** [review]
Looks good to me.

**Review by VeijoPesonen — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -181,6 +181,11 @@ nRF9160 samples
 
     * The sample now waits for RRC idle mode before requesting neighbor cell measurements.
 
+* :ref:`lwm2m_client` sample:
+
+  * Added:
+
+    * Support for 7002 EK shield and Wi-Fi based location.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:188]:
```suggestion
    * Support for nRF7002 EK shield and Wi-Fi based location.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -298,6 +303,13 @@ Libraries for networking
 
   * Fixed a bug where the same buffer was incorrectly shared between caching a P-GPS prediction and loading a new one, when external flash was used.
 
+* :ref:`lib_lwm2m_location_assistance` library:
+
+  * Added:
+
+    * Support for Wi-Fi based location through LwM2M.
+    * API for scanning Wi-Fi Access Points.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:311]:
```suggestion
    * API for scanning Wi-Fi access points.
```

File: include/net/lwm2m_client_utils_location.h [hunk 5]
```diff
@@ -204,6 +204,17 @@ int lwm2m_update_signal_meas_objects(const struct lte_lc_cells_info *const cells
 int lwm2m_ncell_handler_register(void);
 void lwm2m_ncell_schedule_measurement(void);
 
+/**
+ * @brief Request a Wi-Fi scan for nearby access points.
+ *
+ * The Wi-Fi Access Points are added to the Visible Wi-Fi Access Point objects (ID 33627)
```
Comment (peknis) [inline include/net/lwm2m_client_utils_location.h:210]:
```suggestion
 * The Wi-Fi access points are added to the Visible Wi-Fi Access Point objects (ID 33627)
```

File: include/net/lwm2m_client_utils_location.h [hunk 6]
```diff
@@ -204,6 +204,17 @@ int lwm2m_update_signal_meas_objects(const struct lte_lc_cells_info *const cells
...
+ * and used for the Wi-Fi based location when sending a ground fix request.
+ *
+ * @return Returns a negative error code (errno.h) indicating
+ *         reason of failure or 0 for success.
```
Comment (peknis) [inline include/net/lwm2m_client_utils_location.h:214]:
```suggestion
 * @return Returns a negative error code (errno.h) indicating
 *         the reason for failure or 0 for success.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Comment by jarlamsa** [comment]
Thank you for your review @peknis I think I have now addressed all of your comments. Please re-review.

**Review by peknis — APPROVED** [review]

**Pull request merged**